### PR TITLE
session: annotate the previous statement to the error when transaction commit failed

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
The user may want to know the SQL statement in the transaction when commit failed.

### What is changed and how it works?
Annotate the previous statement to the error when transaction commit failed.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
Make an write conflicts case, then search previous statement in log files 
```
[2019/09/09 12:54:12.346 +08:00] [WARN] [session.go:493] ["commit failed"] [conn=3] ["finished txn"="Txn{state=invalid}"] [error="previous statement: update t set j = ? where i = ? [arguments: -3, 3]: [kv:8005]Write conflict, txnStartTS 411043064012079104 is stale [try again later]"] [errorVerbose="[kv:8005]Write conflict, txnStartTS 411043064012079104 is stale [try again later]\nprevious statement: update t set j = ? where i = ? [arguments: -3, 3]"]
[2019/09/09 12:54:12.346 +08:00] [WARN] [session.go:1010] ["run statement error"] [conn=3] [schemaVersion=29] [error="previous statement: update t set j = ? where i = ? [arguments: -3, 3]: [kv:8005]Write conflict, txnStartTS 411043064012079104 is stale [try again later]"] [errorVerbose="[kv:8005]Write conflict, txnStartTS 411043064012079104 is stale [try again later]\nprevious statement: update t set j = ? where i = ? [arguments: -3, 3]"] [session="{\n  \"currDBName\": \"test\",\n  \"id\": 3,\n  \"preparedStmtCount\": 1,\n  \"status\": 2,\n  \"strictMode\": true,\n  \"user\": {\n    \"Username\": \"root\",\n    \"Hostname\": \"127.0.0.1\",\n    \"CurrentUser\": false,\n    \"AuthUsername\": \"root\",\n    \"AuthHostname\": \"%\"\n  }\n}"]
[2019/09/09 12:54:12.346 +08:00] [WARN] [conn.go:669] ["dispatch error"] [conn=3] [connInfo="id:3, addr:127.0.0.1:61812 status:2, collation:utf8_general_ci, user:root"] [sql=COMMIT] [err="[kv:8005]Write conflict, txnStartTS 411043064012079104 is stale [try again later]\nprevious statement: update t set j = ? where i = ? [arguments: -3, 3]"]
```

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: tools/check/go.mod [hunk 1]
```diff
@@ -20,3 +20,5 @@ require (
 	gopkg.in/yaml.v2 v2.2.2 // indirect
 	honnef.co/go/tools v0.0.0-20180920025451-e3ad64cb4ed3
 )
+
+go 1.13
```
Comment (crazycs520) [inline tools/check/go.mod]: remove this?

File: go.mod [hunk 2]
```diff
@@ -75,3 +75,5 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20180531100431-4c381bd170b4
 	sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+go 1.13
```
Comment (crazycs520) [inline go.mod]: remove this?

**Comment by jackysp** [comment]
PTAL @crazycs520

**Comment by jackysp** [comment]
PTAL @lysu

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**